### PR TITLE
Enforce ruff/refurb rules (FURB)

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [x] GitHub Actions have all passed
* [x] Test coverage is 100% (Codecov passes)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -207,30 +207,33 @@ extend-exclude = [
...
     "ANN102",
     "ANN401",
-    "PT004",  # deprecated
-    "PT011",  # TODO: apply this rule
-    "PT012",  # TODO: apply this rule
+    "FURB118", # TODO: apply this rule
+    "FURB140", # TODO: apply this rule
+    "PT004",   # deprecated
+    "PT011",   # TODO: apply this rule
+    "PT012",   # TODO: apply this rule
```
Comment (jhamman) [inline pyproject.toml:236]: do we want to do these here?

**Review by jhamman — COMMENTED** [review]
Looks good here. Just need to resolve the conflict.

File: pyproject.toml (hunk 1 above)
Comment (jhamman) [reply pyproject.toml:236]: I'm just curious if these TODOs are for this PR are a later one.

Comment (DimitriPapadopoulos) [reply pyproject.toml:236]: Let me apply these rules, I'm interested in your opinion.

**Pull request merged**